include/hipSYCL/sycl/libkernel/sscp/builtins/relational.hpp

```diff
@@ -30,7 +30,16 @@
 #ifndef HIPSYCL_SSCP_RELATIONAL_BUILTINS_HPP
 #define HIPSYCL_SSCP_RELATIONAL_BUILTINS_HPP
 
-HIPSYCL_SSCP_BUILTIN bool __hipsycl_sscp_isnan_f32(float x);
-HIPSYCL_SSCP_BUILTIN bool __hipsycl_sscp_isnan_f64(double x);
+#define HIPSYCL_DECLARE_SSCP_GENFLOAT_REL_BUILTIN(name)                        \
+  HIPSYCL_SSCP_BUILTIN bool __hipsycl_sscp_##name##_f32(float);                \
+  HIPSYCL_SSCP_BUILTIN bool __hipsycl_sscp_##name##_f64(double);               \
+
+HIPSYCL_DECLARE_SSCP_GENFLOAT_REL_BUILTIN(isnan)
+
+HIPSYCL_DECLARE_SSCP_GENFLOAT_REL_BUILTIN(isinf)
+
+HIPSYCL_DECLARE_SSCP_GENFLOAT_REL_BUILTIN(isfinite)
+
+HIPSYCL_DECLARE_SSCP_GENFLOAT_REL_BUILTIN(signbit)
```

Comment:
I don't think you actually save a lot by using macros here. You only need the declaration for f32 and f64. The one line that you save per builtin is almost eaten up by the macro definition (plus additional complexity) :-)

Reply:
I don't think we currently use any macros for the SSCP builtin *headers*. If this is the case I think we should not start here (where there's only little benefit) for the sake of consistency.

Reply:
Ok, reverted!